### PR TITLE
adopt newer source generator

### PR DESCRIPTION
**Pull request by tg123**
use new https://github.com/dotnet/roslyn/blob/main/docs/features/incremental-generators.md to generate client code 

now generated model code could work together with json source generator (aot requirement)

### CODE REVIEW

File: src/KubernetesClient/KubernetesClient.csproj [hunk 1]
```diff
@@ -3,6 +3,8 @@
     <PropertyGroup>
         <TargetFrameworks>net6.0;net7.0</TargetFrameworks>
         <RootNamespace>k8s</RootNamespace>
+        <!-- <PublishAot Condition="'$(TargetFramework)' == 'net8.0'">true</PublishAot> -->
```
Comment (brendandburns) [inline src/KubernetesClient/KubernetesClient.csproj:6]: Document why this is commented out or delete?

**Comment by brendandburns** [comment]
Generally looks good, some comments on some dead code/config

File: src/LibKubernetesGenerator/KubernetesClientSourceGenerator.cs [hunk 2]
```diff
@@ -2,146 +2,144 @@
...
+            return builder.Build();
+        }
 
-                if (generators.Contains("model"))
-                {
-                    container.Resolve<ModelGenerator>().Generate(swagger, context);
-                }
+        public void Initialize(IncrementalGeneratorInitializationContext generatorContext)
+        {
+            // System.Diagnostics.Debugger.Launch();
```
[the remark this answers is not in the document]
Comment (brendandburns) [reply src/LibKubernetesGenerator/KubernetesClientSourceGenerator.cs:92]: Please address this comment too.

**Comment by guillaume-chervet** [comment]
hi @tg123 ,

When i use trimming with latest version I obtain at runtime:
Starting in namespace ramiv2-dev-axa-fr
Unhandled exception. System.AggregateException: One or more errors occurred. (Reflection-based serialization has been disabled for this application. Either use the source generator APIs or explicitly configure the 'JsonSerializerOptions.TypeInfoResolver' property.)
---> System.InvalidOperationException: Reflection-based serialization has been disabled for this application. Either use the source generator APIs or explicitly configure the 'JsonSerializerOptions.TypeInfoResolver' property.
at System.Text.Json.ThrowHelper.ThrowInvalidOperationException_JsonSerializerIsReflectionDisabled()
at System.Text.Json.JsonSerializerOptions.ConfigureForJsonSerializer()
at System.Text.Json.JsonSerializerOptions.MakeReadOnly(Boolean)
at System.Text.Json.JsonSerializer.GetTypeInfo(JsonSerializerOptions, Type)
at System.Text.Json.JsonSerializer.GetTypeInfo[T](JsonSerializerOptions)
at System.Text.Json.JsonSerializer.Deserialize[TValue](Stream , JsonSerializerOptions )
at k8s.KubernetesJson.Deserialize[TValue](Stream json, JsonSerializerOptions jsonSerializerOptions)
at k8s.Kubernetes.CreateResultAsync[T](HttpRequestMessage httpRequest, HttpResponseMessage httpResponse, Nullable`1 watch, CancellationToken cancellationToken)
at k8s.AbstractKubernetes.ICoreV1Operations_ListNamespacedPodWithHttpMessagesAsync[T](String namespaceParameter, Nullable`1 allowWatchBookmarks, String continueParameter, String fieldSelector, String labelSelector, Nullable`1 limit, String resourceVersion, String resourceVersionMatch, Nullable`1 sendInitialEvents, Nullable`1 timeoutSeconds, Nullable`1 watch, Nullable`1 pretty, IReadOnlyDictionary`2 customHeaders, CancellationToken cancellationToken)
at k8s.AbstractKubernetes.k8s.ICoreV1Operations.ListNamespacedPodWithHttpMessagesAsync(String namespaceParameter, Nullable`1 allowWatchBookmarks, String continueParameter, String fieldSelector, String labelSelector, Nullable`1 limit, String resourceVersion, String resourceVersionMatch, Nullable`1 sendInitialEvents, Nullable`1 timeoutSeconds, Nullable`1 watch, Nullable`1 pretty, IReadOnlyDictionary`2 customHeaders, CancellationToken cancellationToken)
at k8s.CoreV1OperationsExtensions.ListNamespacedPodAsync(ICoreV1Operations operations, String namespaceParameter, Nullable`1 allowWatchBookmarks, String continueParameter, String fieldSelector, String labelSelector, Nullable`1 limit, String resourceVersion, String resourceVersionMatch, Nullable`1 sendInitialEvents, Nullable`1 timeoutSeconds, Nullable`1 watch, Nullable`1 pretty, CancellationToken cancellationToken)
at SlimFaas.Kubernetes.KubernetesService.ListFunctionsAsync(String kubeNamespace)
at SlimFaas.ReplicasService.SyncDeploymentsAsync(String kubeNamespace)
--- End of inner exception stack trace ---
at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean)
at System.Threading.Tasks.Task.Wait(Int32, CancellationToken)
at System.Threading.Tasks.Task.Wait()
at Program.<Main>$(String[] args)

**Comment by tg123** [comment]
@guillaume-chervet aot is not fully ready, i am still working on it.
reflection is still required

**Comment by guillaume-chervet** [comment]
Is it possible to annotate method with DynamicallyAccessedMembers attribute to make library buildable with trimming activated?
I do not find a solution

**Pull request merged**